kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/InboxServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.motechproject.nms.kilkari.service.impl;
+
+import org.joda.time.Days;
+import org.joda.time.LocalDate;
+import org.motechproject.nms.kilkari.domain.InboxCallDetails;
+import org.motechproject.nms.kilkari.domain.Subscription;
+import org.motechproject.nms.kilkari.domain.SubscriptionPackMessage;
+import org.motechproject.nms.kilkari.domain.SubscriptionStatus;
+import org.motechproject.nms.kilkari.domain.SubscriptionPack;
+import org.motechproject.nms.kilkari.exception.NoInboxForSubscriptionException;
+import org.motechproject.nms.kilkari.repository.InboxCallDetailsDataService;
+import org.motechproject.nms.kilkari.service.InboxService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ *
+ */
+@Service("inboxService")
+public class InboxServiceImpl implements InboxService {
+
+    private static final int DAYS_IN_WEEK = 7;
+
+    private InboxCallDetailsDataService inboxCallDetailsDataService;
+
+    @Autowired
+    public InboxServiceImpl(InboxCallDetailsDataService inboxCallDetailsDataService) {
+        this.inboxCallDetailsDataService = inboxCallDetailsDataService;
+    }
+
+    @Override
+    public long addInboxCallDetails(InboxCallDetails inboxCallDetails) {
+        InboxCallDetails newRecord = inboxCallDetailsDataService.create(inboxCallDetails);
+        return (long) inboxCallDetailsDataService.getDetachedField(newRecord, "id");
+    }
+
+    @Override
+    public SubscriptionPackMessage getInboxMessage(Subscription subscription) throws NoInboxForSubscriptionException {
+
+        if (subscription.getStartDate().isAfter(LocalDate.now())) {
+            // early subscription, play welcome message
+            return null;
+        }
+        if (subscription.getStartDate() == null ||
+                subscription.getStatus() == SubscriptionStatus.DEACTIVATED ||
+                subscription.getStatus() == SubscriptionStatus.PENDING_ACTIVATION) {
+            // there is no inbox for this subscription, throw
+            throw new NoInboxForSubscriptionException(String.format("No inbox exists for subscription %s",
+                    subscription.getSubscriptionId()));
+        }
+
+        SubscriptionPack pack = subscription.getSubscriptionPack();
+        int daysIntoPack = Days.daysBetween(subscription.getStartDate(), LocalDate.now()).getDays();
+        int messageIndex;
+        int currentWeek = daysIntoPack / DAYS_IN_WEEK + 1;
```

Comment:
I understand operator precedence, but could you add parens just to make it explicit